packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -272,11 +261,69 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
     }
   }
 
+  private _calculateMaxHeight() {
+    let { pageSize, options } = this.props;
+    let maxHeight = 0;
+    if (pageSize) {
+      let { selectedIndex } = this.state;
+      if (-1 === selectedIndex) {
+        selectedIndex = 0;
+      }
+
+      let nonSeparatorIndex: number[] = [];
+      let startIndex = selectedIndex;
+      let lookdown = true;
+      let lookup = true;
+      while (nonSeparatorIndex.length < pageSize && (lookdown || lookup)) {
+        if (lookdown) {
+          if (options[startIndex]) {
+            if (options[startIndex].itemType !== DropdownMenuItemType.Divider) {
+              nonSeparatorIndex.push(options[startIndex].index);
+            }
+            startIndex += 1;
+            continue;
+          } else {
+            lookdown = false;
+            startIndex = selectedIndex - 1;
+          }
+        }
+
+        if (lookup) {
+          if (options[startIndex]) {
+            if (options[startIndex].itemType !== DropdownMenuItemType.Divider) {
+              nonSeparatorIndex.push(options[startIndex].index);
+            }
+            startIndex -= 1;
+            continue;
+          } else {
+            lookup = false;
+            break;
+          }
+        }
+      }
+
+      for (let i = 0; i < nonSeparatorIndex.length; i++) {
+        let item = this.refs[OPTION + i] as Element;
+        if (item) {
+          let itemDOM = ReactDOM.findDOMNode(item);
+          if (itemDOM) {
+            maxHeight += itemDOM.clientHeight;
+          }
+        }
+      }
+
+      if (maxHeight !== this.state.calculatedHeight) {
+        this.setState({ calculatedHeight: maxHeight });
+      }
+    }
+  }
+
   // Render separator
   private _renderSeparator(item: IDropdownOption): JSX.Element {
     let { index, key } = item;
     if (index > 0) {
       return <div
+        ref={ OPTION + item.index }
```

Comment:
string refs are legacy and discouraged now https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs